reader.go

```diff
@@ -0,0 +1,72 @@
+package ctxio
+
+import (
+	"context"
+	"io"
+	"sync"
+)
+
+type Reader struct {
+	reader     io.Reader
+	ctx        context.Context
+	progressFn func(int642 int64)
+	mu         sync.RWMutex
+	n          int64
+	err        error
+}
+
+func noonProgressFn(int64) {}
+
+// NewReader creates a new Reader with the given io.Reader
+func NewReader(ctx context.Context, r io.Reader, progressFn func(int642 int64)) *Reader {
```

Comment:
Again not progressFn func(int642 int64). It should be **progressFn func(int64)**